windows-container-samples/msbuild/Dockerfile

```diff
@@ -0,0 +1,15 @@
+FROM microsoft/windowsservercore
+
+SHELL ["powershell", "-Command", "$ErrorActionPreference = 'Stop';"]
+
+RUN (New-Object System.Net.WebClient).DownloadFile('http://download.microsoft.com/download/5/f/7/5f7acaeb-8363-451f-9425-68a90f98b238/visualcppbuildtools_full.exe', 'visualcppbuildtools_full.exe') ; \
+    Start-Process .\visualcppbuildtools_full.exe -ArgumentList '/NoRestart /S' -Wait ; \
+    rm visualcppbuildtools_full.exe
+
+COPY entrypoint.bat entrypoint.bat
+
+SHELL ["cmd"]
+
+WORKDIR /code
+
+ENTRYPOINT ["C:\\entrypoint.bat"]
```

Comment:
Good point. I also prefer your solution setting all environment variables. I found the problem with some of the variables. If a value ends with a backslash `\` then it is stored with a double quote `"` instead. Eg. for the environment variable `WindowsSdkDir`.

Replacing the
```
setx /M $v[0] $v[1]
```
command with 
```
[Environment]::SetEnvironmentVariable($v[0], $v[1], [EnvironmentVariableTarget]::Machine); ```
it works fine. I'll update this PR to get rid of the `entrypoint.bat`.